tests/connection.rs

```diff
@@ -0,0 +1,282 @@
+pub mod stream {
+    use async_graphql::*;
+    use serde_json::{json, Value::Null};
+
+    struct Root;
+
+    #[Object]
+    impl Root {
+        async fn stream_connection(
+            &self,
+            ctx: &Context<'_>,
+            after: Option<Cursor>,
+            before: Option<Cursor>,
+            first: Option<i32>,
+            last: Option<i32>,
+        ) -> FieldResult<Connection<String>> {
+            let stream = futures::stream::iter(vec![
+                "a".to_owned(),
+                "b".to_owned(),
+                "c".to_owned(),
+                "d".to_owned(),
+                "e".to_owned(),
+                "f".to_owned(),
+            ]);
+
+            let mut source = StreamDataSource::new(stream, |s: &String| {
```

Comment:
Is this iterating the entire stream to gather all cursors? If so, it seems inefficient that we'd iterate it twice

Reply:
No, the stream is just mapped with this closure:
https://github.com/async-graphql/async-graphql/pull/100/files#diff-c6747b5500cb9b8d48c79e5d07e7daa9R18